esp/platform/espcommon.cpp

```diff
@@ -0,0 +1,37 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2022 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#include "espcommon.hpp"
+
+IFileIOStream* createBufferedIOStreamWithFileName(const char* fileNameWithPath, IFOmode mode)
+{
```

Comment:
If people agree that this is a useful shared function I think it should go somewhere more widely accessible.  Perhaps jfile.cpp.  

I wouldn't say "WithFileName", but rather "FromFile".

You could combine the buffered and unbuffered into one function with a flag to indicate buffered.

Reply:
For maximum flexibility you can allow the caller to pass in the buffsize parameter.  You could default it to (unsigned)-1 just as createBufferedIOStream() does.

`extern jlib_decl IFileIOStream * createBufferedIOStream(IFileIO * file, unsigned bufsize=(unsigned)-1);
`

Reply:
Moved to jfile as suggested.